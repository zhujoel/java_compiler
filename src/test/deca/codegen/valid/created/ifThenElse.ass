; start main program
; Class Declaration
; Code de la table des méthodes de Object
	LOAD #null, R0
	STORE R0, 4(GB)
	LOAD code.Object.equals, R0
	STORE R0, 5(GB)
; Main Function
; Test pour savoir si la pile est pleine
	TSTO #5
	BOV pile_pleine
	ADDSP #5
; Declaration des variables
	LOAD #2, R2
	STORE R2, 6(GB)
	LOAD #0, R0
	STORE R0, 9(GB)
	LOAD #0x0.0p0, R0
	STORE R0, 11(GB)
; Beginning of main instructions:
; Début du if_1
; ==
	LOAD 6(GB), R2
	PUSH R2
	LOAD #2, R2
	CMP R2, R2
	BNE else_1
	POP R2
; Contenu du if_1
	WSTR "a est égal à deux."
	BRA endif_1
else_1:
	WSTR "a est différent de deux."
endif_1:
	LOAD #0, R2
	STORE R2, 9(GB)
; Début du if_2
	LOAD 9(GB), R2
	CMP #1, R2
	BNE else_2
; Contenu du if_2
	WSTR "b vaut true"
	BRA endif_2
else_2:
	WSTR "b vaut false"
endif_2:
	PUSH R2
	LOAD #0x1.4cccccp2, R2
	STORE R2, 11(GB)
	POP R2
; Début du if_3
; ==
	LOAD 11(GB), R2
	PUSH R2
	LOAD #0x1.4cccccp2, R2
	CMP R2, R2
	BNE else_3
	POP R2
; Contenu du if_3
	WSTR "c vaut 5.2"
	BRA endif_3
else_3:
	WSTR "c est différent de 5.2"
endif_3:
	HALT
pile_pleine:
	WSTR "Erreur : Pile pleine."
	WNL
	ERROR
dereferencement_null:
	WSTR "Erreur : dereferencement null."
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : Tas plein."
	WNL
	ERROR
debordement:
	WSTR "Erreur : debordement durant une operation arithmétique"
	WNL
	ERROR
io_error:
	WSTR "Erreur : input ou output"
	WNL
	ERROR
; end main program
