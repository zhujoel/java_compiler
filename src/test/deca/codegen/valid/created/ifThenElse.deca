{
    int a = 2;
    bool b;
    float c;
    if(a == 2){
        print("a est égal à deux.");
    }
    else{
        print("a est différent de deux.");
    }

    b = false;
    if(b){
        print("b vaut true");
    }
    else{
        print("b vaut false");
    }
    c = 5.2;
    if(c == 5.2){
        print("c vaut 5.2");
    }
    else{
        print("c est différent de 5.2");
    }
}
