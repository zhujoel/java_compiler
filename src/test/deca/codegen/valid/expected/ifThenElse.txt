a est égal à deux.
b vaut false
c vaut 5.2
